llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp

```diff
@@ -1688,6 +1688,13 @@ static const Comdat *getWasmComdat(const GlobalValue *GV) {
 }
 
 static SectionKind getWasmKindForNamedSection(StringRef Name, SectionKind K) {
+  // segments within the data section.
+  // This could be avoided if all data segements (the wasm sense) were
+  // represented as thier own sections (in the llvm sense).
```

Comment:
*their

Reply:
I went ahead and fixed that upstream: https://github.com/llvm/llvm-project/commit/5a0d8c31a32711dd1101ebcd0a563dec13efdfb9

Reply:
Yeah I'm not authoring this code, I'm just cherry-picking from upstream.